app.py

```diff
@@ -41,152 +23,153 @@
-
-# Now split dataframe into START vs. OK (or whatever else)
-df_start = df[df["action"] == "START"]
-df_ok = df[df["action"] != "START"]
-
-# Drop and rename columns before joining
-df_start = df_start.rename(columns={"timecode": "start_time"}).drop(
-    columns=["action", "raw_line", "total_models"]
-)
-df_ok = df_ok.rename(columns={"timecode": "end_time"}).drop(
-    columns=["model_name", "action"]
-)
-
-# And then join based on model_num
-df_joined = df_start.merge(df_ok, on="model_num", how="outer")
-
-# Extract models that are still running and count them
-df_running = (
-    df_joined.query("end_time.isnull()")
-    .sort_values(by=["start_time"])
-    .reset_index(drop=True)
-)
-
-# How many models are still running?
-running_models = df_running.shape[0]
-
-if running_models == 0:
-    st.subheader("Hmm, I couldn't find any models still running! 🤔")
-    st.caption(
-        "What do you think? Drop some feedback in [the repo](https://github.com/foundinblank/dbt-model-finder/) or email me at adamstone@gmail.com."
+    with st.expander("dbt output example"):
+        st.code(EXAMPLE_LOGS)
+
+
+def get_logs():
+    '''
+    Get the dbt output from the user
+    '''
+    # Ask for dbt output
+    st.header("")
+
+    return st.text_area(
+        "Paste your in-progress dbt output here (or copy the example above).",
+        height=300,
+        help="The text should've come from your command line interface, the dbt Cloud UI or the dbt Cloud logs.",
+        placeholder=EXAMPLE_LOGS,
     )
-elif running_models == 1:
-    st.subheader(f"Looks like there's 1 model still running. 🤏")
-    st.dataframe(df_running[["model_num", "start_time", "model_name"]])
-    st.caption(
-        "What do you think? Drop some feedback in [the repo](https://github.com/foundinblank/dbt-model-finder/) or email me at adamstone@gmail.com."
+
+
+def clean_input(raw_input):
+    '''
+    Clean the input and turn it into a dataframe.
+    Input: raw dbt output. Output: pandas dataframe with the following columns:
+    - raw_line: raw line of text from the dbt output
+    - Model Number: model number in the dbt run
+    - Start Time: time the model started
+    - Model Name: name of the model
+    '''
+    # Check if dbt output is entered
+    if not raw_input:
+        st.stop()
+
+    # Process dbt output
+    input = raw_input.splitlines()
+
+    # Remove empty lines and whitespace
+    input = [line for line in input if not line.isspace()]
+
+    # Turn modified input into dataframe
+    df = pd.DataFrame(input, columns=["raw_line"])
+
+    # TO DO: Sort out the status & runtime columns
+    # TO DO: Add functionality for dbt tests and dbt build
+    # Parse out the start time, model number, model name, and status using regex
+    # Throw an error if the input doesn't look like dbt output
+    try:
+        df["post_regex"] = df["raw_line"].apply(
+            lambda col: re.search(REGEX, col).groups()
+        )
```

Comment:
See previous comment about using regex approach, help convince me this is better than `df = df["raw_line"].str.split(expand=True)`!